src/main/scala/cognite/spark/v1/AssetHierarchyBuilder.scala

```diff
@@ -69,6 +69,26 @@ final case class InvalidNodeReferenceException(nodeIds: Seq[String], referencedF
       val refNodes = referencedFrom.map(x => s"'$x'").sorted.mkString(", ")
       s"${plural("Parent", "Parents")} $nodes referenced from $refNodes ${plural("does", "do")} not exist."
     })
+final case class AttemptedRootChangeException(cause: CdpApiException, newSubtreeRoot: Asset)
```

Comment:
Suggestions welcome for a different name here

Reply:
Maybe "Invalid" instead of "Attempted", but I think either choice is ok.

Reply:
I like "invalid", thanks!